gensim/similarity_learning/drmm_tks.py

```diff
@@ -0,0 +1,111 @@
+import keras.backend as K
+from keras.models import Model
+from keras.layers import Input, Embedding, Dot, Dense, Lambda, Reshape, Dropout
+from keras.activations import softmax
+
+
+class DRMM_TKS:
+    """This is a variant version of DRMM, which applied topk pooling in the matching matrix.
+    It has the following steps:
+    1. embed queries into embedding vector named 'q_embed' and 'd_embed' respectively
+    2. computing 'q_embed' and 'd_embed' with element-wise multiplication
+    3. computing output of upper layer with dense layer operation
+    4. take softmax operation on the output of this layer named 'g' and find the k largest entries named 'mm_k'.
+    5. input 'mm_k' into hidden layers, with specified length of layers and activation function
+    6. compute 'g' and 'mm_k' with element-wise multiplication.
+
+    # Returns
+        Score list between queries and documents.
+    """
+
+    def __init__(self, embedding, embed_dim, vocab_size, embed_trainable=False, target_mode='ranking',
+                 topk=20, dropout_rate=0., text_maxlen=40, hidden_sizes=[5, 1]):
+        """Initializes the model
+        Parameters:
+        ----------
+        embedding: numpy array matrix
+            A numpy array matrix which has the embeddings extracted from a pretrained
+            word embedding like Stanford Glove
+            This is fed to the Embedding Layer which then outputs the word embedding
+
+        embed_dim: int
+            The dimension of the above mentioned vectors
+
+        vocab_size: int
```

Comment:
how user should know it?

Reply:
The user  gets the value from the wikiqa extractor

```
wikiqa = WikiQA_DRMM_TKS_Extractor(file_path=file_path, word_embedding_path=word_embedding_path)
drmm_tks = DRMM_TKS(embedding=wikiqa.embedding_matrix, vocab_size=wikiqa.vocab_size + 1, text_maxlen=wikiqa.text_maxlen)
```

Note: `vocab_size` **+ 1** to include pad words

Although, the `vocab_size` can be derived from the `embedding` parameter which has shape `(vocab_size+1, embedding_dim)`, it might make it confusing for the user.
What do you think?